src/jdk.jartool/share/classes/sun/tools/jar/Main.java

```diff
@@ -775,6 +790,9 @@ private void expand(File dir, String[] files, Set<String> cpaths, int version)
         if (files == null)
             return;
 
+        // Ensure files list is sorted for reproducible jar content
+        Arrays.sort(files);
+
```

Comment:
Have you had a chance to measure the performance with a large number of Zip entries with this change?

Reply:
No I haven't, but my thoughts on this were assuming you had a zip with many 1000s of ZipEntries the file I/O would be far more significant. Also, you will note this is not sorting the whole set, just within each directory, so the sort won't be complex, unless you had 1000s of files in a single directory. The "non-determinism" comes from the File.list() implementation which uses OS file listing, whose order is non-deterministic.